manifests/sentinel.pp

```diff
@@ -1,7 +1,42 @@
 # @summary Install redis-sentinel
 #
-# @param auth_pass
-#   The password to use to authenticate with the master and slaves.
+# @param master_name
```

Comment:
Wouldn't it be better to define a type alias using a [struct](https://puppet.com/docs/puppet/latest/lang_data_abstract.html#the-struct-data-type)?

Reply:
you mean something like this:
`type Redis::MasterName = Struct[{`

instead of this:
`type Redis::MasterName = Hash[String, Struct[{`

?

Reply:
What do u think about this:
```
type Redis::MasterName = Struct[{
  name                   => String,
  redis_host             => Stdlib::Host,
  redis_port             => Stdlib::Port,
  quorum                 => Integer[1],
  down_after             => Integer[1],
  parallel_sync          => Integer[0],
  failover_timeout       => Integer[1],
  auth_pass              => Optional[String],
  notification_script    => Optional[Stdlib::Absolutepath],
  client_reconfig_script => Optional[Stdlib::Absolutepath],
}]
```

The puppet code will look like:

```
     master_name   => {
       'session_6381' => {
         name             => 'redis-session-instance-6381',
         redis_host       => $redis_master_ip,
         redis_port       => 6381,
         quorum           => 2,
         parallel_sync    => 1,
         down_after       => 5000,
         failover_timeout => 12000,
         auth_pass        => $redis_auth,
       },
       'cache_6380' => {
         name             => 'redis-cache-instance-6380',
         redis_host       => $redis_master_ip,
         redis_port       => 6381,
         quorum           => 2,
         parallel_sync    => 1,
         down_after       => 5000,
         failover_timeout => 12000,
         auth_pass        => $redis_auth,
       },
```

The template would be changed to:
```
<%- | 
      $name,
      $redis_host,
      $redis_port,
      $quorum,
      $down_after,
      $parallel_sync,
      $failover_timeout,
      $auth_pass,
      $notification_script,
      $client_reconfig_script,
| -%>

sentinel monitor <%= $name%> <%= $redis_host %> <%= $redis_port %> <%= $quorum %>
sentinel down-after-milliseconds <%= $name%> <%= $down_after %>
sentinel parallel-syncs <%= $name%> <%= $parallel_sync %>
sentinel failover-timeout <%= $name%> <%= $failover_timeout %>
<% if $auth_pass { -%>
sentinel auth-pass <%= $name%> <%= $auth_pass %>
<% } -%>
<% if $notification_script { -%>
sentinel notification-script <%= $name%> <%= $notification_script %>
<% } -%>
<% if $client_reconfig_script { -%>
sentinel client-reconfig-script <%= $name%> <%= $client_reconfig_script %>
<% } -%>
```

Do you mean something like that?

Reply:
I didn't realize you already had a struct. I was mostly thinking about the documentation, since you specify all keys. Would it be better to document the type instead?

My suggestion would be to generate the reference (`rake strings:generate:reference`) and see how they both look like. See what makes the most sense.

Reply:
I've changed the variable name to make clear, that we have an breaking change with this feature.
The handling is a little bit more user friendly.

And i run puppet strings to document the data type.